cafebabel/users/models.py

```diff
@@ -36,31 +41,28 @@ def has_role(self, role, or_admin=True):
 
     @property
     def profile(self):
-        return self.profiles.get()
-
+        return UserProfile.objects(user=self).get()
 
-@post_save(sender=User)
-def create_user_profile(ModelClass, instance, created):
-    UserProfile.create(user=instance)
+    @classmethod
+    def post_save(cls, sender, document, **kwargs):
+        if kwargs.get('created', False):
+            # Create the profile only on creation (vs. update).
+            UserProfile(user=document).save()
 
 
-class UserRoles(db.Model):
-    user = ForeignKeyField(User, related_name='roles')
-    role = ForeignKeyField(Role, related_name='users')
-    name = property(lambda self: self.role.name)
-    description = property(lambda self: self.role.description)
+signals.post_save.connect(User.post_save, sender=User)
 
 
-class UserProfile(db.Model):
-    name = CharField(null=True)
-    user = ForeignKeyField(User, related_name='profiles', on_delete='CASCADE')
-    socials = PickledField(null=True)
-    website = CharField(null=True)
-    about = CharField(null=True)
+class UserProfile(db.Document):
```

Comment:
Do we still need this as a document or as a EmbeddedDocument?

Reply:
Good question, I don't know Mongo well enough to choose, wdYt?